src/main/scala/com/gu/ssm/aws/EC2.scala

```diff
@@ -26,13 +26,18 @@ object EC2 {
       .build()
   }
 
+  def makeFilter(tagName: String, values: List[String]) = new Filter(s"tag:$tagName", values.asJava)
+
   def resolveByTags(tagValues: List[String], client: AmazonEC2Async)(implicit ec: ExecutionContext): Attempt[List[Instance]] = {
+    val allTags = tagValues ++ tagValues.map(_.toUpperCase) ++ tagValues.map(_.toLowerCase)
+
+    // if user has provided fewer than 3 tags then assume order app,stage,stack
```

Comment:
The comment doesn't match the tag order below.

```suggestion
    // if user has provided fewer than 3 tags then assume order app,stack,stage
```

Reply:
ah good spot, thanks! This was actually the order I wanted so I've corrected the line below.